haas/deferred.py

```diff
@@ -51,22 +52,33 @@ def apply_networking():
         network = action.new_network
         if nic.port:
             if network:
-                network_map[nic.port.label] = network.network_id
+                network_id = network.network_id
             else:
-                network_map[nic.port.label] = None
+                network_id = None
+
+            switch = nic.port.owner
+            if switch.label not in switch_sessions:
+                switch_sessions[switch.label] = switch.session()
+            switch_sessions[switch.label].apply_networking(action)
         else:
             logging.getLogger(__name__).warn(
                 'Not modifying NIC %s; NIC is not on a port.' %
                 nic.label)
 
-    # Apply them
-    driver_name = config.cfg.get('general', 'driver')
-    driver = importlib.import_module('haas.drivers.' + driver_name)
-    driver.apply_networking(network_map)
+    # Close all of our sessions:
+    for session in switch_sessions.values():
+        session.disconnect()
```

Comment:
At some point, we might want to make it so that we can keep these sessions open between calls of `apply_networking`, but that'll take some refactoring